### PR TITLE
AQL String Functions: Describe caseInsensitive parameter, improve examples and port to @EXAMPLE_AQL

### PR DESCRIPTION
**Pull request by Simran-B**
- `caseInsensitive` parameter not sometimes missing, sometimes in the wrong place.
- Examples ported to `@EXAMPLE_AQL` (3.10 only)
- Added a few more examples, renamed a couple to match convention, removed some that didn't provide additional value. Left example queries involving a View unchanged.
- Modified some descriptions and examples that didn't match actual behavior (likely changed around 3.0 noticed)
- Factored out regular expression syntax description and moved it to the end of the page

https://jenkins.arangodb.biz:3456/view/Documentation/job/arangodb-ANY-examples/1605/

### CODE REVIEW

File: 3.10/aql/functions-string.md [hunk 1]
```diff
@@ -24,186 +24,447 @@ Return the number of characters in *value* (not byte length).
...
+  RETURN CONCAT( [[5, 6], {foo: "bar"}] )
+@END_EXAMPLE_AQL
+@endDocuBlock aqlConcatCompoundTypes_2
+{% endaqlexample %}
+{% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
 
 CONCAT_SEPARATOR()
 ------------------
 
 `CONCAT_SEPARATOR(separator, value1, value2, ... valueN) → joinedString`
```
Comment (Simran-B) [inline 3.10/aql/functions-string.md]: A string that was **joined** or that is **joint**?

**Comment by cw00dw0rd** [comment]
@cla-bot check

**Comment by Simran-B** [comment]
There are issues with backslashes in examples. Might need to double escape...

image

**Comment by cw00dw0rd** [comment]
@cla-bot check

**Comment by Simran-B** [comment]
Backslash escape issues fixed:
image

**Review by ansoboleva — APPROVED** [review]
reviewed

**Pull request merged**